Add doc for openEuler 22.03

Yikun opened the pull request:
We fork the doc from existing vesion`cp -R rocky8 openEuler22.03` and apply https://github.com/openhpc/ohpc/pull/1748/commits/bb0eb6e28cc6f15b361433c61fbf70dafae86300 specific changes:

- Apply openEuler changes in steps.tex
- Add openEuler changes to docs.spec
- Skip doc build because the doc.spec depends on a manual doc tar.

https://github.com/openhpc/ohpc/blob/a12ce6765a90b5ad5a8a1709b6958cbe28b48701/components/admin/docs/SPECS/docs.spec#L20

I validate it by build doc locally:
- [aarch64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051844/aarch64_warewulf_openpbs_steps.pdf)
- [aarch64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051845/aarch64_warewulf_slurm_steps.pdf)
- [x86_64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051850/x86_64_warewulf_openpbs_steps.pdf)
- [x86_64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051855/x86_64_warewulf_slurm_steps.pdf)

Yikun commented:
Fix suse CI error in https://github.com/openhpc/ohpc/pull/1749

Yikun committed:
Init openEuler doc from rocky8

In MacOS: cp -R rocky8 openEuler22.03

Signed-off-by: Yikun Jiang <yikunkero@gmail.com>

Yikun commented:
I will rebase soon

adrianreber commented:
This feels wrong. You probably should not copy files from common. That should be a symbolic link.

Yikun commented:
> This feels wrong. You probably should not copy files from common. That should be a symbolic link.

Yes, I also found this, I'm re-making this PR. : )

Yikun committed:
Apply openEuler specific changes

- Apply openEuler changes in steps.tex
- Add openEuler changes to docs.spec
- Skip doc build because final doc.tar needed

Signed-off-by: Yikun Jiang <yikunkero@gmail.com>

Yikun commented:
Validate doc with https://github.com/openhpc/ohpc/pull/1750 :

- [aarch64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051844/aarch64_warewulf_openpbs_steps.pdf)
- [aarch64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051845/aarch64_warewulf_slurm_steps.pdf)
- [x86_64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051850/x86_64_warewulf_openpbs_steps.pdf)
- [x86_64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051855/x86_64_warewulf_slurm_steps.pdf)

Ready for review.

martin-g commented on docs/recipes/install/common/openeuler_repos.tex:
```diff
@@ -0,0 +1,12 @@
+In addition to the \OHPC{} 
+\iftoggle{isxCAT}{and \xCAT{} package repositories,}{package repository,}
```
There was a discussion recently on the mailing list that xCAT is no more supported. Since 2.3 IIRC.
But it could be cleaned up in a separate PR.

Yikun replied:
![image](https://user-images.githubusercontent.com/1736354/227412832-033ba7de-a4f7-46c2-b7b0-09572f0a8439.png)

Yes, I think we can remove it in separate PR to cleanup `isxCAT` if needed, BTW rendered pdf also didn't include this becuase `isxCAT` is false in openEuler.

Yikun commented:
@adrianreber Would you mind to take a look? I think it's ready to go.

adrianreber commented:
> @adrianreber Would you mind to take a look? I think it's ready to go.

I will probably wait for documentation related changes a bit longer before merging them. From my point of view this is currently blocked by the test system. Once that is setup correctly for 3.x I would merge this PR.

Yikun commented:
@adrianreber OK, thanks!

martin-g commented on components/admin/docs/SPECS/docs.spec:
```diff
@@ -96,6 +96,12 @@ make ; %{parser} steps.tex > recipe.sh ; popd
 pushd docs/recipes/install/leap15/x86_64/warewulf/openpbs
```
I think we need an `%if 0%{?openEuler}` at line 70 to define `source_path` to be `docs/recipes/install/openeuler22.03`

Yikun replied:
Good catch, addressed

martin-g commented:
@mghpcsim This is the PR for openEuler docs that generate the recipe.sh.

I also have a PR in my fork that is based on top of this PR and adds execution of the recipe in a self-hosted Github runner - https://github.com/martin-g/openeuler-ohpc/pull/2 . 
It is not very stable though because the compute nodes sometime need more time to become available after the power reset. Also I wasn't able to find a good way to execute the smoke test (the hello world MPI task) with `prun` without creating a new shell session. I wanted to check how it is being tested currently in Jenkins.

adrianreber commented:
This works in CI and is ready to be merged. There is still a `.DS_Store` part of this PR. Please remove that file.

I think the last two commits can be squashed.

Yikun commented:
@adrianreber done ^

- Remove `.DS_Store` and fix tail space
- Squash last two commits into one.

adrianreber commented:
Thanks for the update.

Do you know why there are packages mentioned with `gnu9` and `arm1`. We do not have the ARM HPC compiler for openEuler and `gnu9` is also not available for 3.x.

Can that be removed or is that content automatically created?

Yikun commented:
As I mentioned in first commit, the openEuler doc is [initialized](https://github.com/openhpc/ohpc/pull/1748/commits/a8f9b725a5da0677644651f8894eb008865f0d1a) from exsiting doc.

Yes, seems we need remove `gnu9`/`arm1-`/`intel-` in openEuler, and we also need to remove `gnu9` for other OS? But frankly, I don't know how to generate these conent automatically.

- According some history like https://github.com/openhpc/ohpc/commit/5fe2484d5badb51a0a1d5c8c8db560f1d42e5308 seems we change `gnu9` to `gnu12` manully.
- And seems there are also some script to help generate these doc: https://github.com/openhpc/ohpc/commits/3.x/docs/recipes/install/common/build_tables.pl

adrianreber commented:
> As I mentioned in first commit, the openEuler doc is [initialized](https://github.com/openhpc/ohpc/pull/1748/commits/a8f9b725a5da0677644651f8894eb008865f0d1a) from exsiting doc.
> 
> Yes, seems we need remove `gnu9`/`arm1-`/`intel-` in openEuler, and we also need to remove `gnu9` for other OS? But frankly, I don't know how to generate these conent automatically.
> 
>     * According some history like [5fe2484](https://github.com/openhpc/ohpc/commit/5fe2484d5badb51a0a1d5c8c8db560f1d42e5308) seems we change `gnu9` to `gnu12` manully.
> 
>     * And seems there are also some script to help generate these doc: https://github.com/openhpc/ohpc/commits/3.x/docs/recipes/install/common/build_tables.pl

Okay. Let's merge this as is and maybe somebody who knows the setup better can help us at some point.
